.github/workflows/pre-release.yml

```diff
@@ -0,0 +1,312 @@
+
+    create-pre-release-branch:
+        name: Create release branch
+        runs-on: ubuntu-latest
+        needs:
+            - test-core
+            - test-stub
+            - test-integrations-common-js-browserfy
+            - test-integrations-common-js-webpack
+            - test-integrations-common-js-rollup
+            - test-integrations-module-js-webpack
+            - test-integrations-module-js-rollup
+            - test-integrations-require-js
+            - confirm-public-repo-master-branch
+        steps:
+            - name: Checkout development branch
+              uses: actions/checkout@v2
+              with:
+                  repository: mparticle/mparticle-web-sdk
+                  ref: development
+
+            - name: Create and push release branch
+              run: |
+                  git checkout -b pre-release/${{ github.run_number }}
+                  git push origin pre-release/${{ github.run_number }}
+
+    pre-release:
+        name: Perform Pre-Release
+        runs-on: ubuntu-latest
+        needs:
+            - create-pre-release-branch
+        env:
+            GITHUB_TOKEN: ${{ secrets.MP_SEMANTIC_RELEASE_BOT }}
+            GIT_AUTHOR_NAME: mparticle-automation
+            GIT_AUTHOR_EMAIL: developers@mparticle.com
+            GIT_COMMITTER_NAME: mparticle-automation
+            GIT_COMMITTER_EMAIL: developers@mparticle.com
+            NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+        steps:
+            - name: Checkout master branch
+              uses: actions/checkout@v3
+              with:
+                  fetch-depth: 0
+                  repository: ${{ github.repository }}
+                  token: ${{ secrets.MP_SEMANTIC_RELEASE_BOT }}
+                  ref: master
+
+            - name: Import GPG Key
+              uses: crazy-max/ghaction-import-gpg@v4
+              with:
+                  gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+                  passphrase: ${{ secrets.GPG_PASSPHRASE }}
+                  git_user_signingkey: true
+                  git_commit_gpgsign: true
+
+            - name: Merge pre-release branch into master branch
+              run: |
+                  git pull origin release/${{ github.run_number }}
+
+            - name: Setup Node.js
+              uses: actions/setup-node@v3
+              with:
+                  node-version: 16.x
+
+            - name: Install dependencies
+              run: npm ci
+
+            - name: Release --dry-run
+              if: ${{ github.event.inputs.dryRun == 'true'}}
+              run: |
+                  npx semantic-release --dry-run
+
+            # NPM Publish happens here
+            - name: Release
+              if: ${{ github.event.inputs.dryRun == 'false'}}
+              run: |
+                  npx semantic-release
+
+            - name: Get Package Version Number
+              uses: dutscher/read-package-json-endpoint-actions@v1.33.7
```

Comment:
nit - 

When searching for this action, I saw a market place action here which had a lot more stars (123 vs 4), https://github.com/marketplace/actions/get-current-package-version.  The behavior is fairly simple, but figured star rating is a good indicator of which to choose when everything else works the same.